src/borg/helpers.py

```diff
@@ -1540,6 +1544,18 @@ def calculate_size(self, item):
     def calculate_csize(self, item):
         return sum(c.csize for c in item.get('chunks', []))
 
+    def calculate_dsize(self, item):
+        chunk_index = self.archive.cache.chunks
+        chunks = item.get('chunks', [])
+        chunks_counter = Counter(c.id for c in chunks)
```

Comment:
Can you also put this new/corrected logic into calculate_unique_chunks above? That way the definition of a unique chunk would be consistent for all of these.